### PR TITLE
Add support for ShadowRoot DOM api feature detect

### PR DESCRIPTION
**Pull request by pascalim**
Add support for ShadowRoot DOM api feature detect through `shadowroot` and `shadowroot-legacy` tests

### CODE REVIEW

**Comment by rejas** [comment]
Thx for the PR. But it test for both the old and the newer ShadowDom Api. The first check is against the deprecated call: https://caniuse.com/#feat=mdn-api_element_createshadowroot
while the later is the favored modern one.
Therefor I guess the test should be split into two, what dou you think?

**Comment by pascalim** [comment]
Good eye, yep sounds good, updated PR with `shadowroot` and `shadowroot-legacy` tests

**Comment by rejas** [comment]
Tests are failing, did you notice? Please fix that first  :-)

**Comment by pascalim** [comment]
@rejas it's a case of me not reading the documentation first :p Ok refactored the tests and it seems to appease the test runners

File: feature-detects/dom/shadowroot.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
+/*!
+{
+  "name": "Shadow DOM API",
+  "property": ["shadowroot", "shadowrootlegacy"],
```
Comment (rejas) [inline feature-detects/dom/shadowroot.js]: please split up again in two files, that way we can get rid of one if we want to deprecate the old one

File: feature-detects/dom/shadowroot.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  "notes": [
+    {
+      "name": "MDN Docs",
+      "href": "https://developer.mozilla.org/en-US/docs/Web/API/ShadowRoot"
+    }
+  ],
+  "authors": ["Kevin Coyle", "Pascal Lim"],
```
Comment (rejas) [inline feature-detects/dom/shadowroot.js]: maybe add your github usernames too: "Pascal Lim (@pascalim)"

File: feature-detects/dom/shadowroot.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  "tags": ["dom"]
+  }
```
Comment (rejas) [inline feature-detects/dom/shadowroot.js]: please add the caniuse references too https://caniuse.com/#feat=shadowdom https://caniuse.com/#feat=shadowdomv1

**Comment by pascalim** [comment]
PR updated with requested changes

**Comment by rejas** [comment]
thx a lot! good job!

**Pull request merged**